src/Microsoft.AspNetCore.WebHooks.Receivers/WebHookReceiverConfig.cs

```diff
@@ -0,0 +1,173 @@
+    /// '<c>https://&lt;host&gt;/api/webhooks/incoming/custom/{id}</c>'.
+    /// </summary>
+    public class WebHookReceiverConfig : IWebHookReceiverConfig
+    {
+        // Character that appears between the default configuration value and the first id / value pair and between
+        // later id / value pairs.
+        private static readonly char[] BetweenPairSeparators = new[] { ',' };
+
+        // Character that appears between the receiver id and configuration value in a pair.
+        private static readonly char[] PairSeparators = new[] { '=' };
+
+        private readonly IDictionary<string, string> _settings;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebHookReceiverConfig"/> which will use the application
+        /// settings set in the given <paramref name="configuration"/>.
+        /// </summary>
+        /// <param name="configuration">
+        /// The <see cref="IConfiguration"/> to read for <see cref="IWebHookReceiver"/> configuration.
+        /// </param>
+        /// <param name="loggerFactory">The <see cref="ILoggerFactory"/>.</param>
+        public WebHookReceiverConfig(IConfiguration configuration, ILoggerFactory loggerFactory)
+        {
+            Configuration = configuration;
+            var logger = loggerFactory.CreateLogger<WebHookReceiverConfig>();
+            _settings = ReadSettings(configuration, logger);
+        }
+
+        /// <inheritdoc />
+        public IConfiguration Configuration { get; }
+
+        /// <inheritdoc />
+        public virtual Task<string> GetReceiverConfigAsync(string configurationName, string id)
+        {
+            if (configurationName == null)
+            {
+                throw new ArgumentNullException(nameof(configurationName));
+            }
+            if (id == null)
+            {
+                id = string.Empty;
+            }
+
+            var key = GetConfigKey(configurationName, id);
+            var result = _settings.TryGetValue(key, out var value) ? value : null;
+
+            return Task.FromResult(result);
+        }
+
+        internal static IDictionary<string, string> ReadSettings(IConfiguration configuration, ILogger logger)
+        {
+            // All keys are lowercased before additions or lookups.
+            var settings = new Dictionary<string, string>(StringComparer.Ordinal);
+
+            foreach (var setting in configuration.AsEnumerable())
+            {
+                var key = setting.Key;
+                if (key.Length > WebHookConstants.ReceiverConfigurationKeyPrefix.Length &&
+                    key.StartsWith(WebHookConstants.ReceiverConfigurationKeyPrefix, StringComparison.OrdinalIgnoreCase))
+                {
+                    // Extract configuration (again, likely receiver) name
+                    var configurationName = key.Substring(WebHookConstants.ReceiverConfigurationKeyPrefix.Length);
```

Comment:
This is interesting, I think this needs some design because stuff like this is pretty high impact and we want it to be consistent across the stack